Wikipedia/UI-V5/WMFArticleViewController.m

```diff
@@ -363,6 +369,37 @@ - (void)updateUIForMode:(WMFArticleControllerMode)mode animated:(BOOL)animated {
     self.headerGalleryViewController.view.userInteractionEnabled = mode == WMFArticleControllerModeNormal;
 }
 
+#pragma mark - Loading message container view
+
+- (void)setupLoadingMessageContainerViewIfNecessary {
+    if (!self.loadingMessageContainerView) {
+        self.loadingMessageContainerView =
+            [[[UINib nibWithNibName:@"WMFArticleViewLoadingView" bundle:nil] instantiateWithOwner:nil options:nil] firstObject];
+        [self.headerView addSubview:self.loadingMessageContainerView];
+        self.headerView.clipsToBounds = NO;
+        CGFloat heightOfSpaceBelowHeaderView = MAX(100, self.tableView.frame.size.height - self.headerView.frame.size.height - self.tableView.scrollIndicatorInsets.top - self.tableView.scrollIndicatorInsets.bottom);
+        [self.loadingMessageContainerView mas_makeConstraints:^(MASConstraintMaker* make) {
+            make.top.equalTo(self.headerView.mas_bottom);
+            make.leading.equalTo(self.headerView.mas_leading);
+            make.trailing.equalTo(self.headerView.mas_trailing);
+            make.height.equalTo(@(heightOfSpaceBelowHeaderView));
+        }];
+    }
+}
+
+- (void)showLoadingMessageContainerView {
+    [self setupLoadingMessageContainerViewIfNecessary];
+    self.loadingMessageContainerView.hidden = NO;
+}
+
+- (void)tableView:(UITableView*)tableView willDisplayCell:(UITableViewCell*)cell forRowAtIndexPath:(NSIndexPath*)indexPath {
+    if (indexPath.section == 0) {
+        if (indexPath.row == 0) {
```

Comment:
can you make this a single `if` statement w/ an `&&`? we're not in tokyo, so no need to rightward drift... 😬